lib/aws/AwsSigv4Signer.js

```diff
@@ -15,6 +15,50 @@ const Transport = require('../Transport');
 const aws4 = require('aws4');
 const AwsSigv4SignerError = require('./errors');
 
+const getAwsSDKCredentialsProvider = async () => {
+  // First try V3
+  try {
+    const awsV3 = await import('@aws-sdk/credential-provider-node');
+    if (typeof awsV3.defaultProvider === 'function') {
+      return awsV3.defaultProvider();
+    }
+  } catch (err) {
+    // Ignore
+  }
+  try {
+    const awsV2 = await import('aws-sdk');
+    if (awsV2.default && typeof awsV2.default.config.getCredentials === 'function') {
+      return () =>
+        new Promise((resolve, reject) => {
+          awsV2.default.config.getCredentials((err, credentials) => {
+            if (err) {
+              reject(err);
+            } else {
+              resolve(credentials);
+            }
+          });
+        });
+    }
+  } catch (err) {
+    // Ignore
+  }
+
+  throw new AwsSigv4SignerError(
+    'Unable to find a valid AWS SDK, please provide a valid getCredentials function to AwsSigv4Signer options.'
+  );
+};
+
+const defaultCredentialsAcquirer = () =>
```

Comment:
Should this be an `awsDefaultCredentialsProvider()`?

Reply:
roger that

Reply:
done